storages/backends/s3.py

```diff
@@ -347,6 +349,9 @@ def __init__(self, **settings):
                 s3={"addressing_style": self.addressing_style},
                 signature_version=self.signature_version,
                 proxies=self.proxies,
+                max_pool_connections=64,  # thread-safe
+                tcp_keepalive=True,
+                retries={"max_attempts": 6, "mode": "adaptive"},
```

Comment:
fyi, these values have been empirically chosen using a c6 family EC2 machine talking directly to s3 from the AWS network using [this code](https://gist.github.com/ddelange/13b0f9da3147f3754b9e1e88c13303ba). increasing threads beyond 64 starts to become unstable and does not yield a total throughput increase.

Reply:
Any idea what the defaults are?

People run on all kinds of hardware and I'm mostly inclined to say this should be configured by the user if they want.

Reply:
iirc the default is no retries, no keepalive, 10 conns. the above is the sensible default imo. people can pass their own client_config if the need arises